### PR TITLE
Update contributing and repo management docs for github first

### PR DESCRIPTION
**Pull request by GMNGeoffrey**

### CODE REVIEW

File: docs/repository_management.md [hunk 1]
```diff
@@ -7,12 +7,14 @@ external contributors, but they are documented here for clarity and
...
-GitHub-first and hide the merging complexity in a separate `google` feature
-branch so that standard development workflows don't have to bear the cost for
-every contribution. During this part of the migration period, please direct PRs
-to the `google` branch (which will be marked as the default branch). See
-https://groups.google.com/d/msg/iree-discuss/F07vsG9Ah4o/uAIusKO-BQAJ.
+## Branches
+
+The default branch is called `main`. PRs should be sent there. We also have a
+`google` branch that is synced from the Google internal source repository. This
+branch is merged from and to `main` frequently to usptream any changes coming
```
Comment (ScottTodd) [inline docs/repository_management.md:14]:
```suggestion
branch is merged from and to `main` frequently to upstream any changes coming
```
typo

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ just a few small guidelines you need to follow.
 ## Contributor License Agreement
 
 Contributions to this project must be accompanied by a Contributor License
-Agreement. You (or your employer) retain the copyright to your contribution;
+Agreement. You (or your employer) retain the copyright to your contribution;
```
Comment (ScottTodd) [inline CONTRIBUTING.md:9]:
```suggestion
Agreement. You (or your employer) retain the copyright to your contribution;
```
trailing space (editor can trim on save)

Comment (GMNGeoffrey) [reply CONTRIBUTING.md:9]: Configured! Thanks for teaching me to fish :-D

File: docs/repository_management.md (hunk 1 above)
Comment (GMNGeoffrey) [reply docs/repository_management.md:14]: Done. Can't seem to get spellright working.

**Pull request merged**